applications/minotari_console_wallet/src/init/mod.rs

```diff
@@ -818,42 +828,65 @@ pub fn prompt_wallet_type(
     non_interactive: bool,
 ) -> Option<WalletType> {
     if non_interactive {
-        return Some(WalletType::Software);
+        return Some(WalletType::default());
     }
 
     if wallet_config.wallet_type.is_some() {
-        return wallet_config.wallet_type;
+        return wallet_config.wallet_type.clone();
     }
 
     match boot_mode {
         WalletBoot::New => {
             #[cfg(not(feature = "ledger"))]
-            return Some(WalletType::Software);
+            return Some(WalletType::default());
 
             #[cfg(feature = "ledger")]
             {
                 if prompt("\r\nWould you like to use a connected hardware wallet? (Supported types: Ledger)") {
                     print!("Scanning for connected Ledger hardware device... ");
-                    let err = "No connected device was found. Please make sure the device is plugged in before
-            continuing.";
-                    match TransportNativeHID::new(&HidApi::new().expect(err)) {
-                        Ok(_) => {
+                    match get_transport() {
+                        Ok(hid) => {
                             println!("Device found.");
                             let account = prompt_ledger_account().expect("An account value");
-                            Some(WalletType::Ledger(account))
+                            let ledger = LedgerWallet::new(account, wallet_config.network, None);
```

Comment:
We need to check the ledger app name and version here before we continue:
```suggestion
        let ledger = LedgerWallet::new(account, wallet_config.network, None);
        // Verify that the user selected the correct application on the Ledger device
        match ledger
            .build_command(Instruction::GetPublicAlpha, vec![])
            .execute_with_transport(&hid)
        {
            Ok(result) => {
                // Verify the response
                ...
            }
            Err(e) => panic!("{}", e),
        }
        match ledger
            .build_command(Instruction::GetVersion, vec![])
            .execute_with_transport(&hid)
        {
            Ok(result) => {
                // Verify the response
                ...
            }
            Err(e) => panic!("{}", e),
        }
```

Reply:
Ticket opened here: https://github.com/orgs/tari-project/projects/16/views/1?pane=issue&itemId=63048614